### PR TITLE
SSCS-8007 - Enhanced confidentiality

### PR DESCRIPTION
**Pull request by ryandac**
SSCS-8007 - Create bundles for cases with enhanced Confidentiality

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java [hunk 1]
```diff
@@ -86,92 +90,193 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
         }
-        sscsCaseData.setMultiBundleConfiguration(configs);
     }
 
-    private boolean checkMandatoryFilesMissing(SscsCaseData sscsCaseData) {
-        if (null != sscsCaseData.getDwpDocuments()) {
+    private boolean hasPhmeRequestOrConfidentialityUnderReview(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response, boolean hasEditedDwpResponseDocument, boolean hasEditedDwpEvidenceBundleDocument) {
+        if (isPhmeStatusUnderReview(sscsCaseData) && (hasEditedDwpResponseDocument || hasEditedDwpEvidenceBundleDocument)) {
+            response.addError("There is a pending PHME request on this case");
+            log.info("There is a pending PHME request on this case");
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java:168]: Log message doesn't really mean anything unless we put a case id with it

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java [hunk 2]
```diff
@@ -86,92 +90,193 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+            return true;
+        }
 
-            List<DwpDocument> dwpResponseDocs = sscsCaseData.getDwpDocuments().stream().filter(e -> DwpDocumentType.DWP_RESPONSE.getValue().equals(e.getValue().getDocumentType())).collect(toList());
-            List<DwpDocument> dwpEvidenceBundleDocs = sscsCaseData.getDwpDocuments().stream().filter(e -> DwpDocumentType.DWP_EVIDENCE_BUNDLE.getValue().equals(e.getValue().getDocumentType())).collect(toList());
+        if (isAtLeastOneRequestInProgress(sscsCaseData)) {
+            response.addError("There is a pending enhanced confidentiality request on this case");
+            log.info("There is a pending enhanced confidentiality request on this case");
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java:174]: Log message doesn't really mean anything unless we put a case id with it

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java [hunk 3]
```diff
@@ -86,92 +90,193 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
-        sscsCaseData.setMultiBundleConfiguration(configs);
     }
 
-    private boolean checkMandatoryFilesMissing(SscsCaseData sscsCaseData) {
-        if (null != sscsCaseData.getDwpDocuments()) {
+    private boolean hasPhmeRequestOrConfidentialityUnderReview(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response, boolean hasEditedDwpResponseDocument, boolean hasEditedDwpEvidenceBundleDocument) {
+        if (isPhmeStatusUnderReview(sscsCaseData) && (hasEditedDwpResponseDocument || hasEditedDwpEvidenceBundleDocument)) {
+            response.addError("There is a pending PHME request on this case");
+            log.info("There is a pending PHME request on this case");
+            return true;
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java:169]: Would it be better to add both errors together (if both exist) so the user knows everything that is wrong with the case?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandlerTest.java [hunk 4]
```diff
@@ -287,12 +279,66 @@ public void givenCaseWithEditedDwpDocsAndPhmeUnderReview_thenReturnErrorMessageA
...
+
+        callback.getCaseDetails().getCaseData().setIsConfidentialCase(YesNo.YES);
+        callback.getCaseDetails().getCaseData().setLanguagePreferenceWelsh(langPreference);
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        verify(serviceRequestExecutor).post(callback, "bundleUrl.com/api/new-bundle");
+        assertEquals(1, response.getData().getMultiBundleConfiguration().size());
+        assertEquals(expectedBundleName, response.getData().getMultiBundleConfiguration().get(0).getValue());
+    }
+
```
Comment (jackmaloney) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandlerTest.java:333]: Do you think it's worth adding a unit test (or an integration test) that checks the scenario for when the case is confidential AND a PHME case?

Comment (ryandac) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandlerTest.java:333]: yes, I missed that.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java (hunk 3 above)
Comment (ryandac) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java]: changed

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**